packages/components/src/ui/create-styles/create-style-system/create-core-element.js

```diff
@@ -0,0 +1,182 @@
+/**
+ * External dependencies
+ */
+import isPropValid from '@emotion/is-prop-valid';
+
+/**
+ * WordPress dependencies
+ */
+import { useMergeRefs } from '@wordpress/compose';
+import { forwardRef, createElement } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useHydrateGlobalStyles } from '../hooks';
+import {
+	INTERPOLATION_CLASS_NAME,
+	REDUCED_MOTION_MODE_ATTR,
+} from './constants';
+import {
+	DEFAULT_STYLE_SYSTEM_OPTIONS,
+	getInterpolatedClassName,
+} from './utils';
+
+const shouldForwardProp = isPropValid;
+
+const defaultOptions = DEFAULT_STYLE_SYSTEM_OPTIONS;
+
+/**
+ * @typedef CreateCoreElementOptions
+ * @property {import('create-emotion').ObjectInterpolation<any>} baseStyles The baseStyles from the Style system.
+ * @property {import('../create-compiler').Compiler} compiler The injectGlobal from the Style system's compiler.
+ * @property {import('./generate-theme').GenerateThemeResults} globalStyles The globalStyles from the Style system.
+ */
+
+/**
+ * Creates the core styled elements for the Style system. These elements are
+ * an in-between of Emotion's <div css="" /> solution and a styled.div``
+ * solution.
+ *
+ * createCoreElement is a super light-weight higher-order wrapper that
+ * enhances base elements (like `div`, `input`, or even Components) with
+ * features provided by the Style system, such as direct CSS compiling via
+ * the `css` prop.
+ *
+ * A styled element also has built-in baseStyles (which can be adjusted using
+ * the createStyleSystem factory).
```

Comment:
How is this useful?

Reply:
For example, here:

https://github.com/WordPress/gutenberg/blob/ea4544d971cb1e355721e192236719e364a06ae3/packages/components/src/ui/create-styles/create-style-system/create-core-element.js#L72-L77

I think that's a good usage of the base-styles, ensuring that all style system components automagically comply with reduced motion.

Reply:
I do agree these styles are good defaults (both border box and reduced motion) but I'm not sure it's worth the complexity specially if it's the only thing added by this function.

Reply:
This function does more than just add the base styles:

* Adds base styles
* Adds `css` function to `styled` components (used by the component system)
* Adds the `interpolationClassname` to `styled` components
* Automatically handles prop filtering so that invalid props passed to `styled` components don't automatically get forwarded even if they're used by the `styled` component in a props function (although we don't use this in practice so I think we should remove it, it's also risky to me anyway)

Reply:
After my most recent push that leaves us with: 

* Adds `css` function to `styled` components (used by the component system)
* Adds the `interpolationClassname` to `styled` components

The `interpolationClassname` is such a powerful and simple feature, we should _not_ sacrifice it. It actually is a pretty good performance benefit to have it as the only alternative solution to it is to use React context which will just add more nodes to the React tree OR to hand-write classnames into CSS-in-JS but that is _not_ a common pattern and should be recommended against strongly unless there's no other way to do things.

The base-styles I'm unattached to, but if we're already doing this for the interpolation classname we might as well leave the base styles too.